Add agent.monitoring.pprof doc (#1265)
elastic/observability-docs: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-monitoring.asciidoc

  logs: true
  # enables metrics monitoring
  metrics: true
  # exposes /debug/pprof/ endpoints

>   # recommended that these endpoints are only enabled if the monitoring endpoint is set to localhost

Reviewer: ```suggestion
  # enable these endpoints if the monitoring endpoint is set to localhost
```

Reviewer: suggest shortening this a bit for skim readers.

To enable monitoring, set `agent.monitoring.enabled` to `true`. Also set the
`logs` and `metrics` settings to control whether logs, metrics, or both are
collected. If neither setting is specified, monitoring is turned off. Set
`use_output` to specify the output to which monitoring events are sent.

The `agent.monitoring.pprof` option controls whether the {agent} exposes the
`/debug/pprof/` endpoints with the monitoring endpoints. It is set to `true`

> by default. It is recommended that these endpoints are disabled on the agent
> if the monitoring endpoint is accessible over a network (not recommended).

Reviewer: ```suggestion
by default. If the monitoring endpoint is accessible over a network (not recommended),
set this option to `false` to disable the `/debug/pprof/` endpoints.
```

Reviewer: Suggest changing sentence order a bit so that it's clear that "not recommended" applies to the first part of the sentence only.

Also...unless there are legit reasons why users would want to leave the endpoints enabled, I would just tell them to set the option to false. 

If there are legit reasons, then change my edit to:

If the monitoring endpoint is accessible over a network (not recommended),
we recommend setting this option to `false` to disable the `/debug/pprof/` endpoints.

One other point: by "is accessible" do you mean "is accessed". It's a subtle difference, but worth clarifying in the sentence if you really mean "is accessed"
